TempleEight/docs/guide/metrics.md

```diff
@@ -3,3 +3,146 @@ id: metrics
 title: Metrics
 sidebar_label: Metrics
 ---
+import useBaseUrl from '@docusaurus/useBaseUrl';
+
+Being able to extract meaningful metrics is a crucial part of any microservice-oriented system.
+That's why we've made it easy to add metrics to your Temple project, for you to view quantitative data about successes and failures, as well as latency information.
+
+For this guide, we'll be using the project from the [Getting Started](../getting-started) guide as our starting point.
+
+## Adding Metrics to Your Project
+To add metrics to your Temple project, you just need to add a single metadata item to your `project`:
+
+```templefile {5}
+ExampleProject: project {
+  #language(go);
+  #database(postgres);
+  #provider(dockerCompose);
+  #metrics(prometheus);
+}
+
+ExampleService: service {
+  foo: string;
+  bar: int;
+}
+```
+
+This indicates that each service in the project will export some metrics to [Prometheus](https://prometheus.io), a popular open source monitoring system.
+Prometheus data is then exposed for visualisation with [Grafana](https://grafana.com/grafana/).
+This is currently the only supported metrics configuration that Temple supports.
+
+After regenerating your project, you'll find several new folders have appeared:
+
+```bash
+.
+├── grafana
+│   └── provisioning
+│       ├── dashboards
+│       │   ├── dashboards.yml
+│       │   └── example-service.json
+│       └── datasources
+│           └── datasource.yml
+└── prometheus
+    └── prometheus.yml
+```
+
+These two folders provide the configuration for Grafana and Prometheus, and are mounted into their associated Docker container when it is started.
+More information about these files can be found in the [Grafana Documentation](https://grafana.com/docs/grafana/latest/reference/dashboard/) and [Prometheus Documentation](https://prometheus.io/docs/prometheus/latest/configuration/configuration/).
+
+## Viewing Metrics
+Viewing metrics from your service is very straight forward, first spin up your infrastructure using the provided deployment script:
+
+```bash
+❯❯❯ source deploy.sh
+```
+
+Then, open `localhost:3000` in your browser.
+
+:::tip
+If you're using Kubernetes instead of Docker Compose, you'll want to run the following command, which will open the Grafana dashboard in your browser:
+
+```
+❯❯❯ minikube service grafana 
+```
+:::
+
+This should present you with the Grafana login window.
+The default login is `admin` for both the username and password:
+
+<img alt="Grafana Login" src={useBaseUrl('img/grafana-login.png')}/>
+
+After logging in, you'll want to head to the left-side bar, and select Dashboards > Manage Dashboards
```

Comment:
Mention that it'll take you to a Page Not Found but not to worry about it

Reply:
This only happens because you're rendering the docs on the same port as Grafana. I'm not actually sure how this works, but if you don't render docs it works fine.